cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/PipelineConfigurer.java

```diff
@@ -31,4 +33,18 @@
    * @return stage configurer
    */
   StageConfigurer getStageConfigurer();
+
+  /**
+   * @return the engine for this pipeline
+   */
+  Engine getEngine();
+
+  /**
+   * Set pipeline properties that will be applied to each run of the pipeline.
+   * Depending on the engine used, the properties will be added to the SparkConf for each run or the mapreduce
```

Comment:
only those that are prefixed. This is meant to mirror whatever we would eventually do with runtime args in cdap, to set things in spark conf for user programs.